packages/dev/core/src/Meshes/abstractMesh.ts

```diff
@@ -1456,14 +1456,14 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
     }
 
     /**
-     * Get the position vertex data and optionally apply skeleton and morphing.
-     * @param applySkeleton defines whether to apply the skeleton
-     * @param applyMorph  defines whether to apply the morph target
-     * @param data defines the position data to apply the skeleton and morph to
-     * @returns the position data
+     * Internal function to get buffer data and possibly apply morphs and normals
+     * @param applySkeleton
+     * @param applyMorph
+     * @param data
+     * @param isNormalData
      */
-    public getPositionData(applySkeleton: boolean = false, applyMorph: boolean = false, data?: Nullable<FloatArray>): Nullable<FloatArray> {
-        data = data ?? Tools.Slice(this.getVerticesData(VertexBuffer.PositionKind));
+    private _getData(applySkeleton: boolean = false, applyMorph: boolean = false, data?: Nullable<FloatArray>, isNormalData: boolean = false): Nullable<FloatArray> {

```

Comment:
You can accept "kind" and pass the right kind to the function (normal or position)

Reply:
Thanks, it does look cleaner this way :D